lib/types.js

```diff
@@ -0,0 +1,59 @@
+const isNumber = val => typeof val === 'number';
+const isString = val => typeof val === 'string';
+const isBoolean = val => typeof val === 'boolean';
+const isArray = val => Array.isArray(val) === true;
+const isObject = val => typeof val === 'object';
```

Comment:
your isObject function needed a little more work. we wouldn't want an array to count as an object.